processing/src/test/java/io/druid/segment/filter/TimeFilteringTest.java

```diff
@@ -0,0 +1,209 @@
+  }
+
+  @Test
+  public void testTimeFilterAsLong()
+  {
+    assertFilterMatches(
+        new SelectorDimFilter(Column.TIME_COLUMN_NAME, "0", null),
+        ImmutableList.<String>of("0")
+    );
+    assertFilterMatches(
+        new SelectorDimFilter(Column.TIME_COLUMN_NAME, "9000", null),
+        ImmutableList.<String>of()
+    );
+
+    assertFilterMatches(
+        new BoundDimFilter(Column.TIME_COLUMN_NAME, "0", "4", false, false, true, null),
+        ImmutableList.<String>of("0", "1", "2", "3", "4")
+    );
+    assertFilterMatches(
+        new BoundDimFilter(Column.TIME_COLUMN_NAME, "0", "4", true, true, true, null),
+        ImmutableList.<String>of("1", "2", "3")
+    );
+
+    assertFilterMatches(
+        new InDimFilter(Column.TIME_COLUMN_NAME, Arrays.asList("2", "4", "8"), null),
+        ImmutableList.<String>of("2", "4")
+    );
+
+    String jsFn = "function(x) { return(x === 3 || x === 5) }";
+    assertFilterMatches(
+        new JavaScriptDimFilter(Column.TIME_COLUMN_NAME, jsFn, null, JavaScriptConfig.getDefault()),
+        ImmutableList.<String>of("3", "5")
+    );
+
+    assertFilterMatches(
+        new RegexDimFilter(Column.TIME_COLUMN_NAME, "4", null),
+        ImmutableList.<String>of("4")
+    );
+
+    assertFilterMatches(
+        new SearchQueryDimFilter(Column.TIME_COLUMN_NAME, new ContainsSearchQuerySpec("2", true), null),
+        ImmutableList.<String>of("2")
+    );
+  }
+
+  @Test
+  public void testTimeFilterWithExtractionFn()
+  {
+    final Map<String, String> stringMap = new HashMap<>();
+    stringMap.put("0", "Monday");
+    stringMap.put("1", "Tuesday");
+    stringMap.put("2", "Wednesday");
+    stringMap.put("3", "Thursday");
+    stringMap.put("4", "Friday");
+    stringMap.put("5", "Saturday");
+    LookupExtractor mapExtractor = new MapLookupExtractor(stringMap, false);
+    LookupExtractionFn exfn = new LookupExtractionFn(mapExtractor, false, "UNKNOWN", false, true);
+
+    assertFilterMatches(
+        new SelectorDimFilter(Column.TIME_COLUMN_NAME, "Monday", exfn),
+        ImmutableList.<String>of("0")
+    );
+    assertFilterMatches(
+        new SelectorDimFilter(Column.TIME_COLUMN_NAME, "Notaday", exfn),
+        ImmutableList.<String>of()
+    );
+
+    assertFilterMatches(
+        new BoundDimFilter(Column.TIME_COLUMN_NAME, "Fridax", "Fridaz", false, false, true, exfn),
+        ImmutableList.<String>of("4")
+    );
+    assertFilterMatches(
+        new BoundDimFilter(Column.TIME_COLUMN_NAME, "Friday", "Friday", true, true, true, exfn),
+        ImmutableList.<String>of()
+    );
+
+    assertFilterMatches(
+        new InDimFilter(Column.TIME_COLUMN_NAME, Arrays.asList("Caturday", "Saturday", "Tuesday"), exfn),
+        ImmutableList.<String>of("1", "5")
+    );
+
+    String jsFn = "function(x) { return(x === 'Wednesday' || x === 'Thursday') }";
+    assertFilterMatches(
+        new JavaScriptDimFilter(Column.TIME_COLUMN_NAME, jsFn, exfn, JavaScriptConfig.getDefault()),
```

Comment:
since we don't automatically convert to strings for javascript, can we add a JavaScriptDimFilter test that operates on the time column values directly?